presto-main/src/main/java/com/facebook/presto/statistics/TrivialStatisticsCalculator.java

```diff
@@ -0,0 +1,181 @@
+            PlanNodeStatistics leftStatistics = context.get(node.getLeft());
+            PlanNodeStatistics rightStatistics = context.get(node.getRight());
+
+            PlanNodeStatistics.Builder joinStatistics = PlanNodeStatistics.builder();
+            if (!leftStatistics.getOutputRowsCount().isValueUnknown() && !rightStatistics.getOutputRowsCount().isValueUnknown()) {
+                double joinOutputRowsCount = Math.min(leftStatistics.getOutputRowsCount().getValue(), rightStatistics.getOutputRowsCount().getValue()) * JOIN_MATCHING_COEEFICIENT;
+                joinStatistics.setOutputRowsCount(new ValueEstimate(joinOutputRowsCount));
+            }
+
+            context.put(node, joinStatistics.build());
+            return null;
+        }
+
+        @Override
+        public Void visitExchange(ExchangeNode node, Map<PlanNode, PlanNodeStatistics> context)
+        {
+            visitChildren(node, context);
+            ValueEstimate exchangeOutputRowsCount = new ValueEstimate(0);
+            for (PlanNode child : node.getSources()) {
+                PlanNodeStatistics childStatistics = context.get(child);
+                if (childStatistics.getOutputRowsCount().isValueUnknown()) {
+                    exchangeOutputRowsCount = ValueEstimate.unknownValue();
+                }
+                else {
+                    exchangeOutputRowsCount = exchangeOutputRowsCount.map(value -> value + childStatistics.getOutputRowsCount().getValue());
+                }
+            }
+
+            PlanNodeStatistics exchangeStatistics = PlanNodeStatistics.builder()
+                    .setOutputRowsCount(exchangeOutputRowsCount)
+                    .build();
+            context.put(node, exchangeStatistics);
+            return null;
+        }
+
+        @Override
+        public Void visitTableScan(TableScanNode node, Map<PlanNode, PlanNodeStatistics> context)
+        {
+            Optional<TableLayoutHandle> layout = node.getLayout();
+            if (!layout.isPresent()) {
+                return super.visitTableScan(node, context);
+            }
+
+            TableStatistics tableStatistics = metadataManager.getTableStatistics(session, node.getTable(), layout.get());
+            PlanNodeStatistics.Builder planNodeStatistics = PlanNodeStatistics.builder();
+            planNodeStatistics.setOutputRowsCount(tableStatistics.getRowsCount());
+            context.put(node, planNodeStatistics.build());
+            return null;
+        }
+
+        @Override
+        public Void visitValues(ValuesNode node, Map<PlanNode, PlanNodeStatistics> context)
+        {
+            ValueEstimate valuesCount = new ValueEstimate(node.getRows().size());
+            PlanNodeStatistics valuesStatistics = PlanNodeStatistics.builder()
+                    .setOutputRowsCount(valuesCount)
+                    .build();
+            context.put(node, valuesStatistics);
+            return null;
+        }
+    }
```

Comment:
add:
 -  `EnforceSingleRow` produces single row
 - `Apply` typically produces the same number of rows as input (no LATERAL JOIN yet)
 - `SemiJoin` at most as left side of join
 - `Aggregation` with empty group by list produces single row